### ISSUE TITLE
Nucleation Glow vs Supermatter Glow

### ISSUE THREAD

**Issue by Rusty-Gravy** [Apollo-Community/ApolloStation#204]
I have noticed that when a Nucleation stands next to a Supermatter the light from the Nucleation overwrites the lighting from the Supermatter.
I'm convinced this is a bug, and shouldn't be too hard to iron out.
Possible reasons:
- That the Nucleation shares the exact same glow as the Supermatter. (Could conflict?)
- The Nucleation doesn't add light to a tile, but instead gives that tile a set value of lighting.
- The Nucleation uses a glow colour that is too close to black.

![lightign](https://cloud.githubusercontent.com/assets/9167646/8149899/554786a2-12cd-11e5-9c34-0c5541e230ed.png)

**Comment by nullbear** [comment]
yes! omg if someone fixes this i will be many happiers

**Comment by stuicey** [comment]
It's probably due to the light values used for nucleation being almost black. The way coloured lights work is they blend all colours in the square together

**Comment by nullbear** [comment]
wouldn't it be better if colored lights used OR logic instead of avg or sum?
Lets say the lighting was rgb values instead of hex
color 1 is "5, 10, 15"
lumin 1 is 5
color 2 is "15, 10, 5"
lumin 2 is 7
The total lighting color would be: "15, 10, 15"  and luminosity = 7, instead of what i assume is currently something like:
color = 10, 10, 10
luminosity = 12

**Comment by Kwask** [comment]
If you wanted colors to be blended in a more realistic manner, you would weigh each RGB value with its source's luminosity before averaging it.

So RGB 10, 0, 10 with luminosity 3 would be weighed to 30, 0, 30
RGB 0, 3, 0 with luminosity 1 would be weighed to 0, 3, 0
And to average the two, you'd sum them and divide by the total luminosity and would end up with a blended color of 8, 1, 8

Just summing the colors by itself would result in ultra blinding white light everywhere

**Comment by nullbear** [comment]
yeah but if you take a floodlight and a flashlight, you dont get a dimmer light... I'm saying use the largest light value, instead of summing or averaging them. The larger and more powerful light will always take hold.

However, using the luminosity to determine how to weight the rgb is a good idea. We would get something like this:

```
col1 = 10, 20, 10
lum1 = 3
col2 = 5, 10, 30
lum2 = 2

newcol = max(col1*lum1, col2*lum2) = 30, 60, 60
newcol = newcol / max(lum1, lum2) = 10, 20, 20
```

This would get rid of the effect nucleations and stuff have on lights. For a flare for example, if you used a brighter light alongside a flare, you would get a brighter red light. By using this method, you would result in a slightly brighter light, with less red.

A nucleation walking next to a flare wouldn't make it brighter or change the colour, but his colored lighting would affect the color slightly.

Because of the way this works, you would no longer be able to carry several flashlights and stack their power on top of eachother, it would instead stay at the same level of lighting.

**Comment by Kwask** [comment]
Will be fixed in version 0.1.0
